application/src/main/java/org/togetherjava/tjbot/features/mediaonly/MediaOnlyChannelListener.java

```diff
@@ -48,22 +48,18 @@ public void onMessageReceived(MessageReceivedEvent event) {
         }
 
         if (messageHasNoMediaAttached(message)) {
-            message.suppressEmbeds(true).flatMap(any -> dmUser(message)).queue((res)->{
-                        message.delete().queue();
-                    },
-                    new ErrorHandler()
-                            .handle(ErrorResponse.CANNOT_SEND_TO_USER,
-                                    (err)->{
-                                        if (event.getAuthor().isBot()) {
-                                            return;
-                                        }
-                                        message.delete().flatMap(any -> warnUser(message)).queue((res)->{
-                                            res.delete().queueAfter(1, TimeUnit.MINUTES);
-                                        },(error)->{
-
-                                        });
-                                    })
-            );
+            message.suppressEmbeds(true).flatMap(any -> dmUser(message)).queue((res) -> {
+                message.delete().queue();
+            }, new ErrorHandler().handle(ErrorResponse.CANNOT_SEND_TO_USER, (err) -> {
+                if (event.getAuthor().isBot()) {
+                    return;
+                }
+                message.delete().flatMap(any -> warnUser(message)).queue((res) -> {
+                    res.delete().queueAfter(1, TimeUnit.MINUTES);
+                }, (error) -> {
+
+                });
+            }));
```

Comment:
no, the flow should go as follows
delete the message
try sending a dm
    if success -> good
    else -> send a message in the channel pinging the user

Reply:
I did try to but the else block is not working out.